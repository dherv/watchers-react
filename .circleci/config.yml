version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
