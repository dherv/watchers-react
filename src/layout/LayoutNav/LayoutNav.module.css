.nav {
  margin: 0 32px;
}
.list {
  display: flex;
}
.link {
  display: block;
  height: 100%;
  padding: 24px;
  color: #fff;
  text-decoration: none;
  text-transform: capitalize;
  font-size: 14px;
  border-bottom: 4px solid transparent;
  cursor: pointer;
  transition: all 0.3s linear;
}
