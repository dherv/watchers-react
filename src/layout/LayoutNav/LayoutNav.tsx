import React, { PureComponent, MouseEvent } from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

interface IProps {}

export default class LayoutNav extends PureComponent<IProps> {
  render() {
    return (
      <Nav>
        <List>
          <Item>
            <Link
              activeStyle={{
                backgroundColor: "#121111",
                borderBottom: "4px solid #2a9d90"
              }}
              to="/movies"
            >
              movies
            </Link>
          </Item>
          <Item>
            <Link
              activeStyle={{
                backgroundColor: "#121111",
                borderBottom: "4px solid #2a9d90"
              }}
              to="/series"
            >
              series
            </Link>
          </Item>
          <Item>
            <Link to="/likes">likes</Link>
          </Item>
          <Item>
            <Link to="/watchlist">watchlist</Link>
          </Item>
          <Item>
            <Link to="/account">account</Link>
          </Item>
        </List>
      </Nav>
    );
  }
}
const Nav = styled.nav`
  margin: 0 4rem;
`;
const List = styled.ul`
  display: flex;
`;
const Item = styled.li``;
const Link = styled(NavLink)`
  display: block;
  height: 100%;
  padding: 2rem;
  color: #fff;
  text-decoration: none;
  text-transform: capitalize;
  border-bottom: 4px solid transparent;
  cursor: pointer;
  transition: all 0.3s linear;
`;
