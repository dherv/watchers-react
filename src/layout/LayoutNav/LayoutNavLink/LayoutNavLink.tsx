import React, { SFC } from "react";
import { NavLink } from "react-router-dom";
import { INavLink } from "../../../types/interfaces";
import styles from "../LayoutNav.module.css";

const LayoutNavLink: SFC<{ link: INavLink }> = ({ link: { text, value } }) => (
  <NavLink
    to={value}
    exact
    className={styles.link}
    activeStyle={{
      backgroundColor: "#121111",
      borderBottom: "4px solid #2a9d90"
    }}
  >
    {text}
  </NavLink>
);

export default LayoutNavLink;
