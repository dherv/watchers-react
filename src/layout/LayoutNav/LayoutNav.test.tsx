import { renderer, shallow, React } from "../../test.config";
import LayoutNav from "./LayoutNav";
import { MemoryRouter } from "react-router";
import LayoutNavLink from "../LayoutNavLink/LayoutNavLink";

describe("LayoutNav", () => {
  const props = {};
  const links = [
    { text: "movies", value: "/movies" },
    { text: "series", value: "/series" },
    { text: "likes", value: "/likes" },
    { text: "watchlist", value: "/watchlist" },
    { text: "account", value: "/account" }
  ];
  const wrapper = shallow(<LayoutNav {...props} />);
  describe("component", () => {
    it("should display one <nav> and one <ul>", () => {
      expect.assertions(2);
      expect(wrapper.find("nav")).toHaveLength(1);
      expect(wrapper.find("ul")).toHaveLength(1);
    });
    it("should display li and LayoutNavLink equal to link.length", () => {
      expect.assertions(2);
      // links are generated in render method and count is 5
      expect(wrapper.find("li")).toHaveLength(5);
      expect(wrapper.find(LayoutNavLink)).toHaveLength(5);
    });
    it("should display first LayoutNavLink with first link infos", () => {
      expect.assertions(5);
      wrapper.find(LayoutNavLink).forEach((link, index) => {
        expect(link.prop("link")).toEqual(links[index]);
      });
    });
  });
  describe("snapshot", () => {
    test("should match", () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <LayoutNav {...props} />
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
