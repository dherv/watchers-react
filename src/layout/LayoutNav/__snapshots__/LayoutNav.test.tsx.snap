// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutNav snapshot should match 1`] = `
<nav
  className="nav"
>
  <ul
    className="list"
  >
    <li>
      <a
        aria-current={null}
        className="link"
        href="/movies"
        onClick={[Function]}
      >
        movies
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="link"
        href="/series"
        onClick={[Function]}
      >
        series
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="link"
        href="/likes"
        onClick={[Function]}
      >
        likes
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="link"
        href="/watchlist"
        onClick={[Function]}
      >
        watchlist
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="link"
        href="/account"
        onClick={[Function]}
      >
        account
      </a>
    </li>
  </ul>
</nav>
`;
